mapbox/mapbox-navigation-android
# Added support for delaying audio focus release after finishing playing queued voice instructions.

Closes [#5264](https://github.com/mapbox/mapbox-navigation-android/issues/5264)

### Description
- Added `VoiceInstructionsPlayerOptions.abandonFocusDelay` option that allows customizing an audio focus release delay.
- Updated `MapboxVoiceInstructionsPlayer` logic to delay audio focus release after playing all queued voice instructions.
- Updated the example app `MapboxVoiceActivity` to test audio focus release delay. (Also added NavigationCamera and recenter button for easier testing).
- Regen Metalava API file for `libnavui-voice` module.
- Minor code cleanup and improvements in `MapboxVoiceInstructionsPlayer` and related classes.

### Testing instructions

1. Play music in a music player app of your choosing.
2. Launch the Example App > Voice Example.
3. Long-press on a map to start navigation. (Hint: Select a destination that will generate a route with maneuvers close together.)
4. Observe the voice instructions player. Notice that it will retain audio focus for the additional 2 seconds after playing instructions and hold the focus if there are multiple instructions to play in a sequence.
